### ISSUE TITLE
Input layer didn't get converted

### ISSUE THREAD

**Issue by AIGirl10** [MarsTechHAN/keras2ncnn#18]
Trying to convert keras model to NCNN.

When tried to convert model, it got converted successfully but convolution layer was the first layer

```
Convolution              conv2d_3                                0 1  conv2d_3_blob 0=16 1=3 2=1 3=1 4=0 5=1 6=432 9=1 11=3 12=1 13=1 
```

Not able to understand why input layer (data) couldn't get converted. Any reasons ?

Please help ! thank you.

@MarsTechHAN

**Comment by MarsTechHAN** [comment]
Hi, the input layer should be converted. The converter should find the InputLayer layer and convert that to ncnn Input layer. 
Can you attach your keras file and a screenshot of the file opened by [Netron](https://github.com/lutzroeder/netron).

**Comment by AIGirl10** [comment]
@MarsTechHAN 

Model link: [here](https://drive.google.com/file/d/1FbusUwrZjSyMF4lur_JqWB5jSODmdhO2/view?usp=sharing)

Netron Graph: [here](https://drive.google.com/file/d/1FKP3gXG8qP9_cvEVFXr63Hu6Ux-g1Ee5/view?usp=sharing)

**Comment by MarsTechHAN** [comment]
Hi, from the decoded model config string, there is no InputLayer inside your model. Without an input layer, there is no way to know your input shape settings. 

However, I am thinking maybe I should add a default input Layer. I will PR that later.

---
Here is the decoded config string in your model
```JSON
{
    "class_name": "Sequential",
    "config": {
        "layers": [
            {
                "class_name": "Conv2D",
                "config": {
                    "activation": "relu",
                    "activity_regularizer": null,
                    "batch_input_shape": [
                        null,
                        64,
                        64,
                        3
                    ],
                    "bias_constraint": null,
                    "bias_initializer": {
                        "class_name": "Zeros",
                        "config": {
                            "dtype": "float32"
                        }
                    },
                    "bias_regularizer": null,
                    "data_format": "channels_last",
                    "dilation_rate": [
                        1,
                        1
                    ],
                    "dtype": "float32",
                    "filters": 16,
                    "kernel_constraint": null,
                    "kernel_initializer": {
                        "class_name": "GlorotUniform",
                        "config": {
                            "dtype": "float32",
                            "seed": null
                        }
                    },
                    "kernel_regularizer": null,
                    "kernel_size": [
                        3,
                        3
                    ],
                    "name": "conv2d_3",
                    "padding": "valid",
                    "strides": [
                        1,
                        1
                    ],
                    "trainable": true,
                    "use_bias": true
                }
            },
            {
                "class_name": "AveragePooling2D",
                "config": {
                    "data_format": "channels_last",
                    "dtype": "float32",
                    "name": "average_pooling2d_3",
                    "padding": "valid",
                    "pool_size": [
                        2,
                        2
                    ],
                    "strides": [
                        2,
                        2
                    ],
                    "trainable": true
                }
            },
            {
                "class_name": "Dropout",
                "config": {
                    "dtype": "float32",
                    "name": "dropout_5",
                    "noise_shape": null,
                    "rate": 0.5,
                    "seed": null,
                    "trainable": true
                }
            },
            {
                "class_name": "Conv2D",
                "config": {
                    "activation": "relu",
                    "activity_regularizer": null,
                    "bias_constraint": null,
                    "bias_initializer": {
                        "class_name": "Zeros",
                        "config": {
                            "dtype": "float32"
                        }
                    },
                    "bias_regularizer": null,
                    "data_format": "channels_last",
                    "dilation_rate": [
                        1,
                        1
                    ],
                    "dtype": "float32",
                    "filters": 32,
                    "kernel_constraint": null,
                    "kernel_initializer": {
                        "class_name": "GlorotUniform",
                        "config": {
                            "dtype": "float32",
                            "seed": null
                        }
                    },
                    "kernel_regularizer": null,
                    "kernel_size": [
                        3,
                        3
                    ],
                    "name": "conv2d_4",
                    "padding": "valid",
                    "strides": [
                        1,
                        1
                    ],
                    "trainable": true,
                    "use_bias": true
                }
            },
            {
                "class_name": "AveragePooling2D",
                "config": {
                    "data_format": "channels_last",
                    "dtype": "float32",
                    "name": "average_pooling2d_4",
                    "padding": "valid",
                    "pool_size": [
                        2,
                        2
                    ],
                    "strides": [
                        2,
                        2
                    ],
                    "trainable": true
                }
            },
            {
                "class_name": "Dropout",
                "config": {
                    "dtype": "float32",
                    "name": "dropout_6",
                    "noise_shape": null,
                    "rate": 0.5,
                    "seed": null,
                    "trainable": true
                }
            },
            {
                "class_name": "Conv2D",
                "config": {
                    "activation": "relu",
                    "activity_regularizer": null,
                    "bias_constraint": null,
                    "bias_initializer": {
                        "class_name": "Zeros",
                        "config": {
                            "dtype": "float32"
                        }
                    },
                    "bias_regularizer": null,
                    "data_format": "channels_last",
                    "dilation_rate": [
                        1,
                        1
                    ],
                    "dtype": "float32",
                    "filters": 64,
                    "kernel_constraint": null,
                    "kernel_initializer": {
                        "class_name": "GlorotUniform",
                        "config": {
                            "dtype": "float32",
                            "seed": null
                        }
                    },
                    "kernel_regularizer": null,
                    "kernel_size": [
                        3,
                        3
                    ],
                    "name": "conv2d_5",
                    "padding": "valid",
                    "strides": [
                        1,
                        1
                    ],
                    "trainable": true,
                    "use_bias": true
                }
            },
            {
                "class_name": "AveragePooling2D",
                "config": {
                    "data_format": "channels_last",
                    "dtype": "float32",
                    "name": "average_pooling2d_5",
                    "padding": "valid",
                    "pool_size": [
                        2,
                        2
                    ],
                    "strides": [
                        2,
                        2
                    ],
                    "trainable": true
                }
            },
            {
                "class_name": "Dropout",
                "config": {
                    "dtype": "float32",
                    "name": "dropout_7",
                    "noise_shape": null,
                    "rate": 0.5,
                    "seed": null,
                    "trainable": true
                }
            },
            {
                "class_name": "Flatten",
                "config": {
                    "data_format": "channels_last",
                    "dtype": "float32",
                    "name": "flatten_1",
                    "trainable": true
                }
            },
            {
                "class_name": "Dense",
                "config": {
                    "activation": "relu",
                    "activity_regularizer": null,
                    "bias_constraint": null,
                    "bias_initializer": {
                        "class_name": "Zeros",
                        "config": {
                            "dtype": "float32"
                        }
                    },
                    "bias_regularizer": null,
                    "dtype": "float32",
                    "kernel_constraint": null,
                    "kernel_initializer": {
                        "class_name": "GlorotUniform",
                        "config": {
                            "dtype": "float32",
                            "seed": null
                        }
                    },
                    "kernel_regularizer": null,
                    "name": "dense_3",
                    "trainable": true,
                    "units": 256,
                    "use_bias": true
                }
            },
            {
                "class_name": "Dropout",
                "config": {
                    "dtype": "float32",
                    "name": "dropout_8",
                    "noise_shape": null,
                    "rate": 0.2,
                    "seed": null,
                    "trainable": true
                }
            },
            {
                "class_name": "Dense",
                "config": {
                    "activation": "relu",
                    "activity_regularizer": null,
                    "bias_constraint": null,
                    "bias_initializer": {
                        "class_name": "Zeros",
                        "config": {
                            "dtype": "float32"
                        }
                    },
                    "bias_regularizer": null,
                    "dtype": "float32",
                    "kernel_constraint": null,
                    "kernel_initializer": {
                        "class_name": "GlorotUniform",
                        "config": {
                            "dtype": "float32",
                            "seed": null
                        }
                    },
                    "kernel_regularizer": null,
                    "name": "dense_4",
                    "trainable": true,
                    "units": 128,
                    "use_bias": true
                }
            },
            {
                "class_name": "Dense",
                "config": {
                    "activation": "softmax",
                    "activity_regularizer": null,
                    "bias_constraint": null,
                    "bias_initializer": {
                        "class_name": "Zeros",
                        "config": {
                            "dtype": "float32"
                        }
                    },
                    "bias_regularizer": null,
                    "dtype": "float32",
                    "kernel_constraint": null,
                    "kernel_initializer": {
                        "class_name": "GlorotUniform",
                        "config": {
                            "dtype": "float32",
                            "seed": null
                        }
                    },
                    "kernel_regularizer": null,
                    "name": "dense_5",
                    "trainable": true,
                    "units": 2,
                    "use_bias": true
                }
            }
        ],
        "name": "sequential_1"
    }
}
```

**Comment by AIGirl10** [comment]
@MarsTechHAN 

Ohh okk Thanks. Since netron showed input layer I thought it's there.

About input size model needs 64×64 . Also adding input in Param file doesn't it affect the number of parameters (first line in .param file)

So maybe correct .Param can be something like this
```
Input                    input_0                                 0 1 input_0_blob 0=1 1=3 2=64 3=64
```

Right ?

Again Thanks for your help.

**Comment by MarsTechHAN** [comment]
You can check the code, the first line is just a constant magic, but the second line is the layer count and blob count which should be increment. ncnn uses dynamic shapes alone for all executions, so you don't need to give a constant input shape here.

**Comment by MarsTechHAN** [comment]
Fix in 9841e26, I will close this issue since the issue has been resolved and no other issue raised.
